pyprind 2.11.3 :snowflake:

ifitchet opened the pull request:
pyprind 2.11.3 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5065]
- dev_url:        https://github.com/rasbt/pyprind
- conda_forge:    https://github.com/conda-forge/PyPrind-feedstock/blob/main/recipe/meta.yaml
- pypi:           n/a
- pypi inspector: https://inspector.pypi.io/project/PyPrind/2.11.3

### Explanation of changes:

- basic build
- upstream tests


[PKG-5065]: https://anaconda.atlassian.net/browse/PKG-5065?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ifitchet committed:
basic build

ifitchet committed:
add upstream tests

lorepirri commented on recipe/meta.yaml:
```diff
@@ -13,22 +15,32 @@ source:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
```
We do not build cross-platform, you can remove these

```suggestion
```

lorepirri commented on recipe/meta.yaml:
```diff
@@ -13,22 +15,32 @@ source:
 
 build:
   number: 1005
```
If we are one version after conda-forge, it's a new version therefore we must have a build number to 0

```suggestion
  number: 0
```

ifitchet replied:
Duh!
Hmm, how do I automate a check?  🤔

skupr-anaconda replied:
A new version => reset build number to 0.

lorepirri commented on recipe/meta.yaml:
```diff
@@ -13,22 +15,32 @@ source:
+    - wheel
   run:
     - python
     - psutil >=3.2.0
```
This is optional since version 2.9.2, and it's not in the `setup.py`, you could move it to `run_constrained`

```suggestion
```

See: https://github.com/search?q=repo%3Arasbt%2Fpyprind%20psutil&type=code

skupr-anaconda replied:
`psutils` is imported here https://github.com/rasbt/pyprind/blob/57d8611ae86cc2cb71d6f1ab973476fc9bea5b7a/pyprind/prog_class.py#L20

boldorider4 replied:
It's technically an optional dependency because of the `try except`. If the build doesn't fail after removal, I'd rather not include it as a `run` dependency. I would also consider adding it to the `run_constrained` section.

ifitchet replied:
Of interest, `psutil` is fine in `run_constrained` but is required for the `monitor` tests.

I'll remove the pinning as there doesn't seem to be any mention of `3.2.0` (or anything else) elsewhere.

skupr-anaconda replied:
Then it can be moved it to `test/required`

lorepirri replied:
They made `psutils ` optional since version `2.9.2`, it's always imported between `try-catch`.

See: https://github.com/rasbt/pyprind/blob/881fa94ef2da4acebcd33686fc258f42ca7066fa/CHANGELOG.md?plain=1#L59-L63

ifitchet replied:
added the pinning back in as it is declared in `requirements.txt`

lorepirri reviewed (commented):
You can remove line 12:
```
fn: {{ name }}-{{ version }}.tar.gz
```

lorepirri commented on recipe/meta.yaml:
```diff
@@ -1,6 +1,8 @@
-{% set version = "2.11.2" %}
-{% set sha256 = "c46cab453b805852853dfe29fd933aa88a2a516153909c695b098e9161a9e675" %}
+# XXX Technically, there is no git tag for 2.11.3 but the sole commit
+# since 2.11.2's CHANGELOG does indicate it is meant to be 2.11.3
```
if you performed a diff with pypi, you could add the commit id that you used here

ifitchet committed:
updates for PR comments

psutil moved to run_constrained (without pinning) and added as a test
requirement

ifitchet committed:
psutil pinning defined in requirements.txt

lorepirri reviewed (approved):
Good job on this one
